src/state/apis/zapper/zapperApi.ts

```diff
@@ -240,7 +303,298 @@ export const zapper = createApi({
         return { data }
       },
     }),
+    getZapperAppsBalancesOutput: build.query<
+      GetZapperAppsBalancesOutput,
+      GetZapperAppsbalancesInput
+    >({
+      queryFn: async (_input, { dispatch, getState }) => {
+        const ReadOnlyAssets = selectFeatureFlag(getState() as any, 'ReadOnlyAssets')
+
+        if (!ReadOnlyAssets)
+          return {
+            data: {
+              userData: [],
+              opportunities: {},
+              metadataByProvider: {},
+            },
+          }
+
+        const maybeZapperV2AppsData = await dispatch(
+          zapperApi.endpoints.getZapperAppsOutput.initiate(),
+        )
+
+        const accountIds = selectWalletAccountIds(getState() as ReduxState)
+
+        const assets = selectAssets(getState() as ReduxState)
+        const evmNetworks = evmChainIds.map(chainIdToZapperNetwork).filter(isSome)
+
+        const addresses = accountIdsToEvmAddresses(accountIds)
+        const url = `/v2/balances/apps`
+        const params = {
+          'addresses[]': addresses,
+          'networks[]': evmNetworks,
+        }
+        const payload = { ...options, params, headers, url }
+        const { data: res } = await axios.request({ ...payload })
+        try {
+          const zapperV2AppsBalancessData = V2AppsBalancesResponse.parse(res)
+
+          const parsedOpportunities = zapperV2AppsBalancessData.reduce<GetZapperAppsBalancesOutput>(
+            (acc, appAccountBalance) => {
+              const appId = appAccountBalance.appId
+              const appName = appAccountBalance.appName
+              const appImage = appAccountBalance.appImage
+              const accountId = toAccountId({
+                chainId:
+                  ZAPPER_NETWORKS_TO_CHAIN_ID_MAP[
+                    appAccountBalance.network as SupportedZapperNetwork
+                  ],
+                account: appAccountBalance.address,
+              })
+
+              // [0] only for debugging, obviously support all products eventually
+              const appAccountOpportunities = (appAccountBalance.products[0]?.assets ?? [])
```

Comment:
use all now? is this comment invalid?

Reply:
https://github.com/shapeshift/web/pull/4504/commits/6e9a2e096bec2b52fdea6aba12b7958653b71007